consumer/src/mode/decoded/deposited_curve.rs

```diff
@@ -0,0 +1,333 @@
+    async fn create_event(
+        &self,
+        event: &DecodedMessage,
+        decoded_consumer_context: &DecodedConsumerContext,
+        deposit_id: &str,
+    ) -> Result<Event, ConsumerError> {
+        // Create the event
+        let event = Event::builder()
+            .id(DecodedMessage::event_id(event))
+            .event_type(EventType::Deposited) // Reuse the same event type
+            .deposit_id(deposit_id.to_string()) // Set deposit_id
+            .block_number(U256Wrapper::try_from(event.block_number)?)
+            .block_timestamp(event.block_timestamp)
+            .transaction_hash(event.transaction_hash.clone())
+            .build();
+
+        event
+            .upsert(
+                &decoded_consumer_context.pg_pool,
+                &decoded_consumer_context.backend_schema,
+            )
+            .await
+            .map_err(ConsumerError::ModelError)
+    }
+
+    /// This function initializes accounts for sender and receiver
+    async fn initialize_accounts(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+    ) -> Result<(), ConsumerError> {
+        // Create or get the sender account
+        get_or_create_account(
+            self.sender.to_string(),
+            decoded_consumer_context,
+        )
+        .await?;
+
+        // Create or get the receiver account
+        get_or_create_account(
+            self.receiver.to_string(),
+            decoded_consumer_context,
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    /// This function handles the creation of a curve deposit
+    pub async fn handle_curve_deposit_creation(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+        event: &DecodedMessage,
+    ) -> Result<(), ConsumerError> {
+        info!("Processing DepositedCurve event: {:?}", self);
+
+        // Initialize accounts
+        self.initialize_accounts(decoded_consumer_context).await?;
+
+        // Get or create the curve vault
+        let curve_vault = self.get_or_create_curve_vault(decoded_consumer_context, event).await?;
+
+        // Create deposit record first
+        let deposit_id = self.create_deposit(decoded_consumer_context, event).await?;
+
+        // Create event with deposit_id
+        self.create_event(event, decoded_consumer_context, &deposit_id).await?;
+
+        // Handle position
+        self.handle_position(decoded_consumer_context, &curve_vault).await?;
+
+        // Create signal
+        self.create_signal(decoded_consumer_context, event, &curve_vault).await?;
+
+        Ok(())
+    }
+
+    /// This function gets or creates a curve vault
+    async fn get_or_create_curve_vault(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+        event: &DecodedMessage,
+    ) -> Result<CurveVault, ConsumerError> {
+        // Get the current share price
+        let current_share_price = decoded_consumer_context
```

Comment:
We have a new sharePriceChanged and sharePriceChangedCurve event -- this is emitted whenever the vault totals change.  We can easily add parameters to these events to include the totals and the deltas if desired, which would save us from having to do any math in the backend for the following situations:


1) Fees flowed into pro rata vault for atom deposit/creation/redeem
2) Fees flowed into pro rata vault from triple deposit / creation / redeem

No fees flow into curve vaults

But this will help the back-end keep track of the total balance / shares in each vault without having to do an RPC call